### PR TITLE
Migrate from deprecated TSLint to ESLint and stabilize UI tests

### PR DESCRIPTION
**Pull request by djelinek**
# Pull Request informations

## Rebase & Merge default requirements

1. Green build for main branch
2. Wait 24 hours after PR creation
3. Green job for PR
4. Approved PR

## PR labels default process

- READY_FOR_REVIEW  &rightarrow;  REVIEW_DONE  &rightarrow;  READY_FOR_MERGE

## Tests

- [ ] Are there **Unit tests**?
- [ ] Are there **Integration tests**?
- [ ] Do we need a new **UI test**?

## PR workflow progress

1. [ ] Tagged with relevant **PR labels**
2. [ ] Green **job for PR**
3. [ ] PR was created more than **24 hours ago** or **All committers approved** it
4. [ ] Green **main** branch build

### CODE REVIEW

File: .github/workflows/vscode-versions-matrix.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        vscode-version: ["1.61.2", "1.66.0"]
+        vscode-version: ["1.65.2", "1.66.2"]
```
Comment (apupier) [inline .github/workflows/vscode-versions-matrix.yml:15]: Do we really need to upgrade the version?
it might cause troubles with Theia/Che

Comment (djelinek) [reply .github/workflows/vscode-versions-matrix.yml:15]: ok thanks for explanation! 👍 

I agree we should keep 1.61.2 I will change that back, thanks for catching that!

**Comment by apupier** [comment]
my bad, I missed this one to review and now there is again a conflict. Sorry

**Comment by djelinek** [comment]
PR is ready for review

File: package-lock.json [hunk 2]
```diff
@@ -1,33 +1,7326 @@
 {
   "name": "vscode-apache-camel",
   "version": "0.4.0",
-  "lockfileVersion": 1,
+  "lockfileVersion": 2,
```
Comment (apupier) [inline package-lock.json:4]: would have been better to keep the same version but well, will be boring to redo the change and risk is low

**Review by apupier — APPROVED** [review]
great to move away from long-time deprecated tslint

**Pull request merged**